File: repos/firecrawl/apps/java-sdk/src/main/java/com/firecrawl/errors/RateLimitException.java
package com.firecrawl.errors;

/**
 * Thrown when the API returns a 429 Too Many Requests response.
 */
public class RateLimitException extends FirecrawlException {

    public RateLimitException(String message) {
        super(message, 429);
    }

    public RateLimitException(String message, String errorCode, Object details) {
        super(message, 429, errorCode, details);
    }
}

File: repos/firecrawl/apps/dot-net-sdk/Firecrawl/Exceptions/RateLimitException.cs
namespace Firecrawl.Exceptions;

/// <summary>
/// Thrown when the API returns a 429 Too Many Requests response.
/// </summary>
public class RateLimitException : FirecrawlException
{
    public RateLimitException(string message, string? errorCode = null, object? details = null)
        : base(message, 429, errorCode, details)
    {
    }
}
